src/main/java/com/twilio/rest/api/v2010/account/CallReader.java

```diff
@@ -321,20 +285,12 @@ private void addQueryParams(final Request request) {
             request.addQueryParam("Status", status.toString());
         }
 
-        if (absoluteStartTime != null) {
-            request.addQueryParam("StartTime", absoluteStartTime.toString(Request.QUERY_STRING_DATE_TIME_FORMAT));
-        } else if (rangeStartTime != null) {
-            request.addQueryDateTimeRange("StartTime", rangeStartTime);
-        }
-
-        if (absoluteEndTime != null) {
-            request.addQueryParam("EndTime", absoluteEndTime.toString(Request.QUERY_STRING_DATE_TIME_FORMAT));
```

Comment:
This removes functionality where I could before provide a single `EndTime` rather than needing to provide both upper and lower bound on the end time.

Reply:
I will add a function that replaces that functionality.

Reply:
You can still call `addQueryParam` as above; however, calling `addQueryDateTimeRange` should be transparent to the caller since if only `endTime` is defined, we achieve the same effect as calling `addQueryParam`. 

Are you proposing that we create an additional function that sits on top of `addQueryParam`, something along the lines of `addQueryEndDateTime`? Or in these cases, just use `addQueryParam` as before?

Reply:
No, I'm saying I used to be able to call `setStartTime` and it would add the query param `StartTime=`. Now it does `EndTime>=` I'm guessing?

Reply:
[This](https://github.com/twilio/twilio-java/blob/a5f7797dd4fab4a5f2f0f26d755e6c0431c13538/src/test/java/com/twilio/http/RequestTest.java#L121) is the current behavior when we just set the `startTime` and [this example](https://github.com/twilio/twilio-java/blob/a5f7797dd4fab4a5f2f0f26d755e6c0431c13538/src/test/java/com/twilio/http/RequestTest.java#L130) for setting just the `endTime`.

Reply:
Those are `Request` examples. I'm talking about the `CallReader` interface.

Reply:
Ah, how about replacing `setStartTime()` and `setEndTime()` with `setAbsoluteStartTime()`, `setAbsoluteEndTime()`, `setRangeStartTime()` and `setRangeEndTime()`?

Reply:
Why can't we use what all the other languages use? Examples: https://www.twilio.com/docs/sms/api/message-resource?code-sample=code-read-list-messages-filter-by-before-sent-date&code-language=Ruby&code-sdk-version=5.x

`setStartTime`, `setStartTimeBefore`, `setStartTimeAfter`.

Reply:
The biggest difference is that in other languages we can pass named parameters. In the closest implementation ([C#](https://github.com/twilio/twilio-csharp/blob/main/src/Twilio/Rest/Api/V2010/Account/CallResource.cs#L485)), we have [a builder](https://github.com/twilio/twilio-csharp/blob/main/src/Twilio/Rest/Api/V2010/Account/CallOptions.cs#L457) that sorts the date/time stuff.

In the Java case, we can keep the building of the query params in [`addQueryParams`](https://github.com/twilio/twilio-java/blob/main/src/main/java/com/twilio/rest/api/v2010/account/CallReader.java#L324) and then add the variables `startTime`, `startTimeBefore`, `startTimeAfter`, `endTimeBefore`, `endTime` and `endTimeAfter` along with their getters & setters. How does that implementation sound?

Reply:
That's fine. As long as users can still individually provide `startTime`, `startTimeBefore`, and `startTimeAfter`.